src/plugins/intel_gpu/src/plugin/remote_context.cpp

```diff
@@ -132,48 +132,54 @@ void RemoteBlobImpl::allocate() {
 
-    switch (m_mem_type) {
-    case BlobType::BT_BUF_INTERNAL: {
-        m_memObject = m_engine->allocate_memory(m_layout, cldnn::allocation_type::cl_mem);
-        break;
-    }
-    case BlobType::BT_USM_HOST_INTERNAL: {
-        m_memObject = m_engine->allocate_memory(m_layout, cldnn::allocation_type::usm_host);
-        break;
-    }
-    case BlobType::BT_USM_DEVICE_INTERNAL: {
-        m_memObject = m_engine->allocate_memory(m_layout, cldnn::allocation_type::usm_device);
-        break;
-    }
-    case BlobType::BT_BUF_SHARED: {
-        m_memObject = m_engine->share_buffer(m_layout, m_mem);
-        break;
-    }
-    case BlobType::BT_USM_SHARED: {
-        m_memObject = m_engine->share_usm(m_layout, m_mem);
-        break;
-    }
+    try {
+        switch (m_mem_type) {
+        case BlobType::BT_BUF_INTERNAL: {
+            m_memObject = m_engine->allocate_memory(m_layout, cldnn::allocation_type::cl_mem);
+            break;
+        }
+        case BlobType::BT_USM_HOST_INTERNAL: {
+            m_memObject = m_engine->allocate_memory(m_layout, cldnn::allocation_type::usm_host);
+            break;
+        }
+        case BlobType::BT_USM_DEVICE_INTERNAL: {
+            m_memObject = m_engine->allocate_memory(m_layout, cldnn::allocation_type::usm_device);
+            break;
+        }
+        case BlobType::BT_BUF_SHARED: {
+            m_memObject = m_engine->share_buffer(m_layout, m_mem);
+            break;
+        }
+        case BlobType::BT_USM_SHARED: {
+            m_memObject = m_engine->share_usm(m_layout, m_mem);
+            break;
+        }
 #ifdef _WIN32
-    case BlobType::BT_SURF_SHARED: {
-        m_memObject = m_engine->share_surface(m_layout, m_mem, m_plane);
-        break;
-    }
-    case BlobType::BT_DX_BUF_SHARED: {
-        m_memObject = m_engine->share_dx_buffer(m_layout, m_mem);
-        break;
-    }
+        case BlobType::BT_SURF_SHARED: {
+            m_memObject = m_engine->share_surface(m_layout, m_mem, m_plane);
+            break;
+        }
+        case BlobType::BT_DX_BUF_SHARED: {
+            m_memObject = m_engine->share_dx_buffer(m_layout, m_mem);
+            break;
+        }
 #else
-    case BlobType::BT_SURF_SHARED: {
-        m_memObject = m_engine->share_surface(m_layout, m_surf, m_plane);
-        break;
-    }
+        case BlobType::BT_SURF_SHARED: {
+            m_memObject = m_engine->share_surface(m_layout, m_surf, m_plane);
+            break;
+        }
 #endif
-    case BlobType::BT_IMG_SHARED: {
-        m_memObject = m_engine->share_image(m_layout, m_mem);
-        break;
-    }
-    default:
-        m_memObject.reset();
+        case BlobType::BT_IMG_SHARED: {
+            m_memObject = m_engine->share_image(m_layout, m_mem);
+            break;
+        }
+        default:
+            m_memObject.reset();
+        }
+    } catch (const std::exception& e) {
+        _impl->release_lock();
+        std::cerr << e.what() << '\n';
+        throw e;
```

Comment:
This fix is simple, but there might be another issue in other places, too.

1. What about using tbb spin_lock instead of custom lock?
2. What about implementing BasicLockable and use lock_guard instead? https://en.cppreference.com/w/cpp/named_req/BasicLockable

Reply:
Updated it.